cobra/core/Model.py

```diff
@@ -230,9 +308,85 @@ def add_reactions(self, reaction_list):
 
         self.reactions += reaction_list
 
+        # from cameo ...
+        self._populate_solver(reaction_list)
+
+    def _populate_solver(self, reaction_list, metabolite_list=None):
+        """Populate attached solver with constraints and variables that
+        model the provided reactions.
+        """
+        constraint_terms = AutoVivification()
+        if metabolite_list is not None:
```

Comment:
Thanks for walking me through that though! Realky appreciate it. I think for now we could pull it like that and try make our suggestions for refactoring via pull requests to devel-2.